### ISSUE TITLE
The static import statement shows  "Resolve error: Missing binary"

### ISSUE THREAD

**Issue by midorizemi** [microsoft/vscode-eslint#1175]
vscode-eslint is working my react application project on working directory.
It, however, fails to find fibers package via vscode. I cant find a solution yet. please help me.

Here is my repository. https://github.com/midorizemi/react-sample.git

My environments:
- Windows10 64bit
- nvm v1.1.7 via scoop
- node v14.15.4
- npm v6.14.10

I installed npm packages, and open vscode, and add ESLint plugin in vscode.
When I typed "import" first line, vscode-eslint said the error.
![スクリーンショット 2021-02-04 150421](https://user-images.githubusercontent.com/927671/106851894-6490cb00-66fa-11eb-9f13-96088b410238.jpg)

I, however, tried to run `npm run fix:lint`. I didn't get any error like the error.
```sh
PS C:\Users\Public\react\sample-react> npm run fix:lint

> webpack-react-demo@ fix:lint C:\Users\Public\react\sample-react
> eslint --fix "./{lib,src,test}/**/*.{ts,tsx}"
```

This is the full infomation of the error from "PROBLEMS" form vscode.

```
{
	"resource": "/c:/Users/Public/react/sample-react/src/main.tsx",
	"owner": "eslint",
	"code": {
		"value": "import/extensions",
		"target": {
			"$mid": 1,
			"external": "https://github.com/benmosher/eslint-plugin-import/blob/v2.22.1/docs/rules/extensions.md",
			"path": "/benmosher/eslint-plugin-import/blob/v2.22.1/docs/rules/extensions.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Resolve error: Missing binary. See message above.\n    at Object.<anonymous> (C:\\Users\\Public\\react\\sample-react\\node_modules\\fibers\\fibers.js:23:9)\n    at Module._compile (internal/modules/cjs/loader.js:1014:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1051:10)\n    at Module.load (internal/modules/cjs/loader.js:862:32)\n    at Module._load (internal/modules/cjs/loader.js:774:14)\n    at Function.Module._load (electron/js2c/asar.js:769:28)\n    at Module.require (internal/modules/cjs/loader.js:899:19)\n    at require (internal/modules/cjs/helpers.js:74:18)\n    at Object.<anonymous> (C:\\Users\\Public\\react\\sample-react\\webpack.config.js:60:24)\n    at Module._compile (internal/modules/cjs/loader.js:1014:30)",
	"source": "eslint",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}
```
Although I delete node_modules and reinstall npm packages, that doesn't work. 
thank you.

**Comment by dbaeumer** [comment]
This works for me:

<img width="813" alt="capture" src="https://user-images.githubusercontent.com/1931590/106863406-cb958c00-66c8-11eb-9104-bd84931d2eb4.png">

Are you sure your system is setup so that binaries can be complied on `npm install`

**Comment by dbaeumer** [comment]
And what happens if you execute `./node_modules/.bin/eslint src/main.tsx` in a terminal.

**Comment by midorizemi** [comment]
Thank you so much for quick response.

> Are you sure your system is setup so that binaries can be complied on npm install

yes, I am. And I've uninstall nvm from scoop, and, install nodejs-lts(14.15.4) via scoop.
Then I've executed `npm install`, and I've was ensure that was completed.

I've executed `./node_modules/.bin/eslint src/main.tsx`, and It didn't put anything.
![image](https://user-images.githubusercontent.com/927671/106886081-d121bf00-6726-11eb-94a7-f1d05b653062.png)

I've reopend VSCode, however, It didn't work.
![image](https://user-images.githubusercontent.com/927671/106886193-f6163200-6726-11eb-9ff0-1f2821db35b9.png)

And I've reinstalled vscode-eslint and uninstalled the other, then I replaced setting.json above. 

```json
{
    "[javascript,javascriptreact,javascriptreact,typescriptreact,json]": {
        "editor.tabSize": 2,
    },
    "[javascriptreact]": {
        "editor.formatOnSave": true,
        "editor.codeActionsOnSave": {
            "source.fixAll.eslint": true
        },
    },
    "typescript.updateImportsOnFileMove.enabled": "always",
    "javascript.updateImportsOnFileMove.enabled": "always",
    "typescript.format.insertSpaceAfterTypeAssertion": true,
    "eslint.alwaysShowStatus": true,
    "eslint.probe": [
        "javascript",
        "javascriptreact",
        "typescript",
        "typescriptreact",
        "html",
        "vue",
        "markdown",
        "react"
    ],
    "eslint.workingDirectories": [
        { "mode": "auto" }
    ],
    "terminal.integrated.env.windows": {
        "PATH": "${workspaceRoot}\\node_modules\\.bin;${env:PATH}"
    },
}
```
However, It still didn't work..

**Comment by dbaeumer** [comment]
Wild guess: can you use node as the runtime for eslint by using the setting `eslint.runtime`. Could be the the node shipped with Electron is not capable of executing / loading the binary.

**Comment by midorizemi** [comment]
It works! Thank you so much!
I added the settings `eslint.runtime` to setting.json as follows and restarted VS Code.

```json
"eslint.runtime": "C:\\Users\\midorizemi\\scoop\\apps\\nodejs-lts\\14.15.4\\node.exe",
```
The runtime path is "Environment:" from `scoop info nodejs-lts`.

I don't understand yet why the node included in Electron cannot executing / loading the binary, although, This issue was solved.
Thank you so much for your support again, @dbaeumer!

**Comment by dbaeumer** [comment]
The problem is that when you install the package it is complied against the node version you have installed (which in you case is 14.15.4). Electron ships node version 12.18.3 and the binary compiled for 14.15.4 can't be loaded / executed in 12.18.3.

I know, not something a user usually wants to deal with :-)

**Comment by midorizemi** [comment]
Thank you for additional information.
I can understand the cause finally.
But Can I ask a question?
When I used VSCode in other OS environments (Ubuntu, etc.), I installed 12.18.3 version of node to develop, and in that environment, I was able to load it without any problems like this.
I'm sorry I don't have that knowledge at all.
Does this mean that the version of node provided by Electron is different due to the difference in distributions?

**Comment by dbaeumer** [comment]
No it is not. The problem is that binaries are node version dependent. So if you install 12.18.3 it will work since the binaries are built against the right node API surface.

**Comment by midorizemi** [comment]
Oh. I understand. Thank you so much.

**Comment by midorizemi** [comment]
Due to the fact that I moved my development environment from Windows to Mac M1, I encountered the same problem again.

I had set `eslint.runtime` as `"eslint.runtime":"/Users/midorizemi/.anyenv/envs/nodenv/shims/node",` but eslint showed the same error as the issue. However, eslint showed the same error as the issue.

I was not able to explore the issue in depth, but it seems that .anyenv shims in `"eslint.runtime"` cannot select and run the version of node according to the `.node-versions` in the project directory.

Finally, I worked around this problem by explicitly specifying the version of the executable specified in the project.

```json
// folder/setting.json
"eslint.runtime": "/Users/midorizemi/.anyenv/envs/nodenv/versions/14.17.1/bin/node",
```

This may not be relevant to this project, but I hope it is helpful.
